syntax/scan.go

```diff
@@ -227,8 +227,14 @@ func readSource(filename string, src interface{}) (data []byte, err error) {
 	case string:
 		data = []byte(src)
 	case []byte:
+		if src == nil {
```

Comment:
See my comment on the Issue: a nil value of type []byte is a valid representation of an empty file.  When you assign []byte(nil) to a variable of type interface{}, the result is a non-nil interface and this triggers the parser behavior of reading from the (empty) byte slice, not the named file.